ArduCopter/mode_land.cpp

```diff
@@ -82,8 +82,16 @@ void ModeLand::gps_run()
             land_pause = false;
         }
 
-        // run normal landing or precision landing (if enabled)
-        land_run_normal_or_precland(land_pause);
+        if (!copter.failsafe.radio && (g.throttle_behavior & THR_BEHAVE_HIGH_THROTTLE_CANCELS_LAND) != 0 && copter.rc_throttle_control_in_filter.get() > LAND_CANCEL_TRIGGER_THR){
+            LOGGER_WRITE_EVENT(LogEvent::LAND_CANCELLED_BY_PILOT);
+            // exit land if throttle is high
+            copter.set_mode(Mode::Number::LOITER, ModeReason::THROTTLE_LAND_ESCAPE);
+
+        } else {
```

Comment:
Thanks for this, I think we should remove this else.  I'm just slightly concerned that there could possibly be a reason that the vehicle can't enter Loiter mode but we keep trying over and over and end up never running the regular land_run_normal_or_preclan() method